File: scripts/data-update/fix-extensao-area/data-models/ResponseData/DTOs/UserDto.cs
using data_models.ResponseData.Models.AccessControl;

namespace data_models.ResponseData.DTOs
{
    public class UserDto
    {
        public int Id { get; set; }
        public string? Username { get; set; }
        public bool IsComite { get; set; }
        public bool IsDPO { get; set; }
        public bool IsSystem { get; set; }
        public Group OriginGroup { get; set; }
        public ICollection<Group> Groups { get; set; }
        public string Nome { get; set; }

        public override string ToString()
        {
            return
                $"{nameof(Id)}: {Id}, {nameof(Username)}: {Username}, {nameof(IsComite)}: {IsComite}, {nameof(IsDPO)}: {IsDPO}, {nameof(IsSystem)}: {IsSystem}, {nameof(OriginGroup)}: {OriginGroup}, {nameof(Groups)}: {Groups}, {nameof(Nome)}: {Nome}";
        }
    }
}

File: CPTM.ILA.Frontend/src/shared/models/DTOs/user-dto.ts
import { emptyGroup, Group } from "../access-control/group.model";

export interface UserDto {
    id: number;
    username: string;
    isComite: boolean;
    isDPO: boolean;
    isSystem: boolean;
    originGroup: Group;
    groups: Group[];
    groupAccessExpirationDate: Date;
    nome: string;
}

export const emptyUserDto = (): UserDto => {
    return {
        id: 0,
        username: "",
        isComite: false,
        isDPO: false,
        isSystem: false,
        originGroup: emptyGroup(),
        groups: [],
        groupAccessExpirationDate: new Date(),
        nome: "",
    };
};
